app/main.py

```diff
@@ -1,2 +1,107 @@
+                    break
+                if self.ls[index][0] == key:
+                    self.ls[index] = (key, value, key_hash)
+                    break
+                index += 1
+        else:
+            self.ls[index] = (key, value, key_hash)
+            self.size += 1
+
+    def resize(self):
+        self.capacity *= 2
+        copied_ls = self.ls.copy()
+        self.ls = [None] * self.capacity
+        self.size = 0
+        for cell in copied_ls:
+            if cell is not None:
+                self.add_element(cell[0], cell[1])
+
+    def __getitem__(self, key):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is None:
+            raise KeyError("No value with such key in this dict.")
+
+        while self.ls[index] is not None:
+            if index == self.capacity:
+                index = 0
+            if self.ls[index][2] == key_hash and self.ls[index][0] == key:
+                return self.ls[index][1]
+            index = (index + 1) % self.capacity
+
+    def __len__(self):
+        return self.size
+
+    def clear(self):
+        for index, element in enumerate(self.ls):
+            if element is not None:
+                self.ls[index] = None
+        self.size = 0
+
+    def __delitem__(self, key):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is None:
+            raise KeyError("No value with such key in this dict.")
+
+        while self.ls[index] is not None:
+            if index == self.capacity:
+                index = 0
+            if self.ls[index][2] == key_hash and self.ls[index][0] == key:
+                self.ls[index] = None
+                self.size -= 1
+            index = (index + 1) % self.capacity
+
+    def get(self, key):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is None:
+            raise KeyError("No value with such key in this dict.")
+
+        while self.ls[index] is not None:
+            if index == self.capacity:
+                index = 0
+            if self.ls[index][2] == key_hash and self.ls[index][0] == key:
+                return self.ls[index][1]
+            index = (index + 1) % self.capacity
+
+    def pop(self, key):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is None:
+            raise KeyError("No value with such key in this dict.")
+
+        while self.ls[index] is not None:
+            if index == self.capacity:
+                index = 0
+            if self.ls[index][2] == key_hash and self.ls[index][0] == key:
+                pop = self.ls[index][1]
+                self.ls[index] = None
+                self.size -= 1
+                return pop
+            index = (index + 1) % self.capacity
+
+    def update(self, dictionary):
+        for key, value in dictionary.items():
+            self.add_element(key, value)
```

Comment:
I agree with Taras. In this implementation `self.size` can achieve `self.capacity`. Rework.